airbyte-integrations/connectors/source-sftp-bulk/source_sftp_bulk/source.py

```diff
@@ -45,23 +45,50 @@ def _generate_json_schema(self, dtypes: Dict[str, Any]) -> Dict[str, Any]:
 
         return json_schema
 
-    def _infer_json_schema(self, config: Mapping[str, Any], connection: SFTPClient) -> Dict[str, Any]:
+    def _infer_json_schema(
+        self,
+        config: Mapping[str, Any],
+        connection: SFTPClient,
+    ) -> Dict[str, Any]:
         file_pattern = config.get("file_pattern")
-        files = connection.get_files(config["folder_path"], file_pattern)
+
+        files = connection.get_files(
+            config["folder_path"],
+            file_pattern,
+            max_level=config.get("max_depth", 0),
+        )
 
         if len(files) == 0:
             logger.warning(f"No files found in folder {config['folder_path']} with pattern {file_pattern}")
             return self._default_json_schema
 
         # Get last file to infer schema
         # Use pandas `infer_objects` to infer dtypes
-        df = connection.fetch_file(fn=files[-1], file_type=config["file_type"], separator=config.get("separator"))
+        column_names = config.get("column_names", None)
+        column_names_separator = config.get("column_names_separator", None)
+        header_names = None
+
+        if column_names and column_names_separator:
+            if column_names == "_None_":
```

Comment:
Why are you expecting `_None_` as string here?

Reply:
I'd say this is bad form from my side, and I should update it once I get a hold of how to test it. My idea is to define a special keyword that will tell pandas to autogenerate column names if none is passed.